Playbooks/playbook-DomainTools_Domain_Auto_Enrichment.yml

```diff
@@ -0,0 +1,170 @@
+id: DomainTools_Domain_Auto_Enrichment
```

Comment:
Hi @idovandijk, thanks again for your help. I'm the responsible PM here at DT.

This playbook (and some others below e.g. Guided_Pivot_Enrichment, Tags) are specific use cases that DT users commonly execute within an IR. The default auto-extract may not work specifically in parsing our somewhat extensive data-format, due to it being extensive and complex. With this playbook, we wanted to give the user an option to further extend the Indicator table (a small set of custom fields are added at installation time) to make this possible.

I'm glad to hear fo the auto-extract capability, however not been successful in adding to the Indicator from the War-room. Also, thinking of instances when the enrichment is not performed on an existing Indicator.

Lastly, there are automation scripts that were written to support this for the user. These 2 playbooks will significantly reduce customization/ development time for our user-base and in parsing through our extensive dataset.

Hope this helps. I will let @ChuckWoodraska respond back to the remaining items. I just wanted to ensure I'm able to color the requirements here.

Reply:
Hi again. I'm happy to be of help. I will comment on the other playbooks separately.
As for this playbook, I feel like I should clear things up a bit:
As you said, a dependency of using your specific script that adds the indicator in this playbook, is custom fields at installation time, so we cannot release it as an out-of-the-box playbook that all customers will be using (as it won't work for them, unless they explicitly added those fields). As for the data retention - the way we data related to the investigation is through context. If the data returned from `!domain` is of use to you during an investigation, you shouldn't bother adding it all to the indicator data. The data returned from `!domain` will remain in context. The indicator stores specific information in some pre-existing fields, such as when the indicator was first seen, its score and a couple more.

Can you please elaborate on "not been successful in adding the indicator from the war-room"?
I will explain the flow. Whether the indicator existed before or not, once a URL or an email is involved in an incident in any way, auto-extract will extract the domain from it, and run `!domain` using all active supporting integrations (including DomainTools Iris). This will add all the data returned from `!domain` to the context. The user can then utilize this data to their needs. Subplaybooks can have this data as their inputs to work further on that data too. Furthermore, the indicator will automatically be stored in the "Indicators" page in Demisto. There is no need to actually store that information manually in the indicator itself.

Just an example would be your pivoting playbook. An investigation involving an interesting domain can take place, and then running your pivoting table, which will take the data returned from `!domain`, can use it to further enrich the data.

If you wish to retain the logic of that playbook due to the "last enriched" check,  then it is possible for you to wrap that in the code, or add another command such as `domaintools-enrich-if-outdated` (or something of the sort. Remember that you can still create this playbook locally for yourself, but I still was not convinced that this is an important playbook that all customers should be provided.

As a test for auto-extract, you can go to the war-room of any incident (or the playground). Use the `Print` command, just to print a URL of interest. Set the `auto-extract` setting to `inline`, which means the command will complete once auto-extract finished doing its job. Then look at your context and see that you have all the data returned from `!domain`.
![image](https://user-images.githubusercontent.com/43602124/67282206-e6df8680-f4d9-11e9-9d9a-55349ecc891f.png)
Also, make sure your integration is enabled, otherwise you definitely won't have any data returned.

I hope this clarifies things a bit. If there are additional questions here or if you think I misunderstood, feel free to ask!